pilot/pkg/kube/inject/mesh.go

```diff
@@ -133,6 +133,12 @@ containers:
     valueFrom:
       fieldRef:
         fieldPath: status.podIP
+  - name: ISTIO_META_ISTIO_PROXY_VERSION
+    {{ if eq .IstioProxyVersion "" -}}
```

Comment:
I don't think this will work - or is needed. Once upgrade happens, auto-inject or inject will put the right proxy version and proxy.

@mandarjog - is this code still used ? My understanding was that we now read the config map and this is dead code to be cleaned up ?